pkg/allocation/converters/converter_test.go

```diff
@@ -216,7 +216,7 @@ func TestConvertAllocationRequestToGameServerAllocation(t *testing.T) {
 		},
 		{
 			name:     "all fields are set",
-			features: fmt.Sprintf("%s=false&%s=false&%s=false", runtime.FeaturePlayerAllocationFilter, runtime.FeatureStateAllocationFilter, runtime.FeatureCountsAndLists),
```

Comment:
This test will need to be changed, since it had `TestConvertAllocationRequestToGameServerAllocation` set to false, so it's expecting a `nil` GameServer.

Reply:
Do I need to include something like the below code snippet?
		{
			name: "nil object",
			in:   nil,
			want: nil,
		},

Reply:
Here is the test implementation: https://github.com/googleforgames/agones/blob/e7d01dc55659249cdf33d644ba12402a7fefa2d5/pkg/allocation/converters/converter_test.go#L565-L575

This is a really good example of Go table driven tests: https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Have a look, lemme know if you have questions, and we can go from there 👍🏻

Reply:
The tests were failing with `GameServerState: nil`, but they succeeded after updating it to `GameServerState: &ready`